base/reduce.jl

```diff
@@ -168,10 +168,12 @@ foldr(op, itr) = mapfoldr(identity, op, itr)
 ## reduce & mapreduce
 
 function mapreduce_impl(f, op, A::AbstractArray, ifirst::Integer, ilast::Integer, blksize::Int=pairwise_blocksize(f, op))
-    if ifirst + blksize > ilast
+    if ifirst == ilast
+        @inbounds return r_promote(op, f(A[ifirst]))
```

Comment:
AFAIU  `r_promote()` method defined in Base does not call `op`. But the line in question matches what `_mapreduce()` does in `n==1` case.